src/testFixtures/groovy/grails/plugin/geb/ContainerGebRecordingExtension.groovy

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2024 original author or authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package grails.plugin.geb
+
+import grails.testing.mixin.integration.Integration
+import groovy.transform.CompileStatic
+import groovy.transform.TailRecursive
+import groovy.util.logging.Slf4j
+import org.spockframework.runtime.extension.IGlobalExtension
+import org.spockframework.runtime.model.SpecInfo
+
+import java.time.LocalDateTime
+
+/**
+ * A Spock Extension that manages the Testcontainers lifecycle for a {@link grails.plugin.geb.ContainerGebSpec}
+ *
+ * @author James Daugherty
+ * @since 5.0
+ */
+@Slf4j
+@CompileStatic
+class ContainerGebRecordingExtension implements IGlobalExtension {
+    WebDriverContainerHolder holder
+
+    @Override
+    void start() {
+        holder = new WebDriverContainerHolder(new RecordingSettings())
+        addShutdownHook {
+            holder.stop()
+        }
+    }
+
+    @Override
+    void visitSpec(SpecInfo spec) {
+        if (isContainerGebSpec(spec)) {
```

Comment:
We could do `if (isContainerGebSpec(spec) && validateContainerGebSpec(spec))`. Return boolean true if `validateContainerGebSpec` succeeds.

Reply:
The only validation is making sure it has the `@Integration`  annotation and that annotation must be present if it's a `ContainerGebSpec` for the test to work.  If the validation was optional I would agree, but the method would only ever return true, so I don't see the benefit of in-lining it.  

I only extracted it into a method to help self document what that code was doing.  Would you prefer I inline the method or keep it as validate?

Reply:
It's perfect that you extracted the validation. To me it just feels like it belongs in the condition to proceed. Either it throws or returns true. As it is now, I feel, it is unclear what will happen with this method invocation.